3.0/m_xlinetools.cpp

```diff
@@ -0,0 +1,602 @@
+			if ((negate && match) || (!negate && !match))
+			{
+				i = safei;
+				continue;
+			}
+
+			// Set (time ago): Prefix '-' means less than; no prefix means more than; both match exact (to the second)
+			if (!args.set.empty())
+			{
+				long set = ServerInstance->Time() - ServerInstance->Duration(args.set);
+
+				if ((args.set[0] == '-' && xline->set_time < set) ||
+				    (args.set[0] != '-' && xline->set_time > set))
+				{
+					i = safei;
+					continue;
+				}
+			}
+
+			// Duration: Prefix '+' means longer than; '-' means shorter than; no prefix means exact; '0' matches no expiry
+			if (!args.duration.empty())
+			{
+				unsigned long duration = ServerInstance->Duration(args.duration);
+
+				if ((xline->duration == 0 && args.duration != "0") ||
+				    (args.duration[0] == '+' && xline->duration <= duration) ||
+				    (args.duration[0] == '-' && xline->duration >= duration) ||
+				    (args.duration.find_first_not_of("+-") == 0 && xline->duration != duration))
+				{
+					i = safei;
+					continue;
+				}
+			}
+
+			// Expires (time ahead): Prefix '+' means more than; no prefix means less than; both match exact (to the second)
+			if (!args.expires.empty())
+			{
+				unsigned long expires = ServerInstance->Time() + ServerInstance->Duration(args.expires);
+
+				if ((xline->duration == 0) ||
+				    (args.expires[0] == '+' && xline->set_time + xline->duration < expires) ||
+				    (args.expires[0] != '+' && xline->set_time + xline->duration > expires))
+				{
+					i = safei;
+					continue;
+				}
+			}
+
+			matched++;
+
+			// Skip the rest when just counting matches
+			if (count)
+			{
+				i = safei;
+				continue;
+			}
+
+			std::string expires;
+			const std::string display = xline->Displayable();
+			const std::string duration = (xline->duration == 0 ? "permanent" : InspIRCd::DurationString(xline->duration));
+			const std::string reason = xline->reason;
+			const std::string settime = ServerInstance->TimeString(xline->set_time);
+			if (xline->duration == 0)
+				expires = "doesn't expire";
+			else
+				expires = "expires on " + ServerInstance->TimeString(xline->expiry) + " (in " + InspIRCd::DurationString(xline->expiry - ServerInstance->Time()) + ")";
+
+			std::string ret;
+			if (remove && ServerInstance->XLines->DelLine(display.c_str(), linetype, ret, user))
+			{
+				ServerInstance->SNO->WriteToSnoMask('x', "%s removed %s on %s: %s", user->nick.c_str(),
+					BuildTypeStr(linetype).c_str(), display.c_str(), reason.c_str());
+			}
+			else if (!remove)
+			{
+				user->WriteNotice(InspIRCd::Format("%s on %s set by %s on %s, duration '%s', %s: %s",
+					BuildTypeStr(linetype).c_str(), display.c_str(), xline->source.c_str(),
+					settime.c_str(), duration.c_str(), expires.c_str(), reason.c_str()));
+			}
+
+			i = safei;
+		}
+	}
+
+	bool HandleCmd(LocalUser* user, const Criteria& args, Command* cmd)
+	{
+		bool count = (cmd->name == "XCOUNT" ? true : false);
```

Comment:
Using the ternary operator here is unnecessary as the == operator returns a boolean value you can dirctly assign to count/etc.